src/main/java/seedu/cafectrl/data/OrderList.java

```diff
@@ -74,21 +79,29 @@ public void printOrderList(Menu menu, Ui ui) {
     /**
      * Aggregates orders by updating quantities and total order costs for the same dish.
      *
-     * @param order           The Order object to be aggregated.
+     * @param order The Order object to be aggregated.
      * @param aggregatedOrders The ArrayList of aggregated orders.
      */
     private void aggregateOrder(Order order, ArrayList<Order> aggregatedOrders) {
         logger.info("Aggregating order...");
 
         if (order.getIsComplete()) {
             int index = getIndexByDishName(aggregatedOrders, order.getDishName());
-            aggregatedOrders.get(index)
-                    .setQuantity(aggregatedOrders.get(index).getQuantity() + order.getQuantity());
-            aggregatedOrders.get(index)
-                    .setTotalOrderCost(aggregatedOrders.get(index).getTotalOrderCost() + order.getTotalOrderCost());
+            //if dish is not found in aggregated orders, add the dish into it
+            if (index == -1) {
+                aggregatedOrders.add(new Order(order.getOrderedDish(), order.getQuantity(), order.getTotalOrderCost(),
+                        true));
+            } else {
+                //else add the quantities and totalCost accordingly
+                aggregatedOrders.get(index)
+                        .setQuantity(aggregatedOrders.get(index).getQuantity() + order.getQuantity());
+                aggregatedOrders.get(index)
+                        .setTotalOrderCost(aggregatedOrders.get(index).getTotalOrderCost() + order.getTotalOrderCost());
+            }
         }
     }
 
+    //@@author Shanice Tang
```

Comment:
perhaps remove the whitespace as the github user tag should be one word
```suggestion
    //@@author ShaniceTang
```